### PR TITLE
Add logger to healthcheck endpoint

### PR DESCRIPTION
**Pull request by adrianmroz**

### CODE REVIEW

File: src/server/routes/health/health.mocha.ts [hunk 1]
```diff
@@ -66,19 +69,32 @@ describe("health router", () => {
...
 
-    singleClusterTests.forEach(({ scenario, status, initialized, delay, expectedStatus }) => {
+    singleClusterTests.forEach(({ scenario, status, initialized, delay, expectedStatus, expectedLoggerCalled }) => {
       it(`returns ${expectedStatus} with ${scenario}`, testComplete => {
         mockLoadStatus(wikiBrokerNock, { status, initialized, delay });
         supertest(app)
           .get("/")
-          .expect(expectedStatus, testComplete);
+          .expect(expectedStatus, () => {
+            expect(logSpy.calledOnce).to.be.eq(expectedLoggerCalled);
```
Comment (adrianmroz) [inline src/server/routes/health/health.mocha.ts]: Is this idiomatic way to check this spy?

File: src/server/routes/health/health.mocha.ts [hunk 2]
```diff
@@ -66,19 +69,32 @@ describe("health router", () => {
...
+    afterEach(() => {
+      logSpy.restore();
+    });
+
     const singleClusterTests = [
-      { scenario: "healthy broker", status: 200, initialized: true, delay: 0, expectedStatus: 200 },
-      { scenario: "unhealthy broker", status: 500, initialized: false, delay: 0, expectedStatus: 503 },
-      { scenario: "uninitialized broker", status: 200, initialized: false, delay: 0, expectedStatus: 503 },
-      { scenario: "timeout to broker", status: 200, initialized: true, delay: 200, expectedStatus: 503 }
+      { scenario: "healthy broker", status: 200, initialized: true, delay: 0, expectedStatus: 200, expectedLoggerCalled: false },
```
Comment (adrianmroz) [inline src/server/routes/health/health.mocha.ts]: Should we assert message contents verbatim there? Or just call to logger?

File: src/server/routes/health/health.mocha.ts (hunk 1 above)
Comment (piter75) [reply src/server/routes/health/health.mocha.ts]: Idiomatic or not it is unreadable now
I would totally discard testing if we are really logging in health handler

File: src/server/routes/health/health.ts [hunk 3]
```diff
@@ -83,20 +87,18 @@ const checkDruidCluster = (cluster: Cluster): Promise<ClusterHealth> => {
...
-      .map(clusterHealth => (clusterHealth.status))
-      .reduce(healthStatusReducer, ClusterHealthStatus.healthy);
+const emitHealthStatus = (clusterHealths: ClusterHealth[], response: Response) => {
+  const unhealthyClusters = clusterHealths.filter(({ status }) => status === ClusterHealthStatus.unhealthy);
+  unhealthyClusters.forEach(({ message, host }: ClusterHealth) => {
+    LOGGER.log(`Unhealthy cluster host: ${host}. Message: ${message}`);
+  });
 
-    const httpState = statusToHttpStatusMap[overallHealth];
+  const isSomeUnhealthy = unhealthyClusters.length > 0;
```
Comment (piter75) [inline src/server/routes/health/health.ts]: This is much more readable than healthStatusReducer 👍

**Pull request merged**